test/cppsuite/test_harness/workload_generator.h

```diff
@@ -68,44 +69,56 @@ class workload_generator : public component {
      *      - Open a cursor on each collection.
      *      - Insert m key/value pairs in each collection. Values are random strings which size is
      * defined by the configuration.
+     *      - Store in memory the created collections and the generated keys that were inserted.
      */
     void
-    populate()
+    populate(database &database)
     {
         WT_CURSOR *cursor;
         WT_SESSION *session;
         wt_timestamp_t ts;
-        int64_t collection_count, key_count, value_size;
-        std::string collection_name, config, generated_value, home;
+        int64_t collection_count, key_count, key_cpt, key_size, value_size;
+        std::string collection_name, config, home;
+        std::vector<std::string> collection_names;
+        key_value_t generated_key, generated_value;
         bool ts_enabled = _timestamp_manager->is_enabled();
 
         cursor = nullptr;
-        collection_count = key_count = value_size = 0;
-        collection_name = "";
+        collection_count = key_count = key_size = value_size = 0;
 
         /* Get a session. */
         session = connection_manager::instance().create_session();
         /* Create n collections as per the configuration and store each collection name. */
         testutil_check(_config->get_int(COLLECTION_COUNT, collection_count));
         for (int i = 0; i < collection_count; ++i) {
             collection_name = "table:collection" + std::to_string(i);
-            testutil_check(session->create(session, collection_name.c_str(), DEFAULT_TABLE_SCHEMA));
+            collection_names.push_back(collection_name);
```

Comment:
I'd rather do something like `database._collections[collection_name] = {}`. And then use database.get_collections in the next loop.